### ISSUE TITLE
[68000/010] iGames closes itself or crashes when game title is clicked

### ISSUE THREAD

**Issue by arti040** [MrZammler/iGame#234]
There is a problem with thumbnails loading on plain 68000/010 CPUs (without FPU). If I turn thumbnails off in settings, everything is fine.

Attaching files from my system - MUI classes (68000 version, as mentioned in README), libs - render and guigfx.library (NoFPU versions as well) and iGame executable affected by this issue.

[iGame-bug-report.zip](https://github.com/MrZammler/iGame/files/13289208/iGame-bug-report.zip)

**Comment by walkero-gr** [comment]
@arti040 Thank you so much for your report. I need to know:
1. which version of iGame are you using?
2. please check the settings and tell me if you use the Gfx Library or not. This library needs a very specific version for 68000 cpu. Please check the amigaguide to find it out.
3. If you are not using the Gfx Library, do you have datatypes set up in your system properly? To see if they work as they should, try to load an image from the games folder with multiview.
4. if you do not have screenshots in your games collection, and you only have icons, please make sure you have a datatype installed that can load them. You will find more info in the amigaguide

**Comment by arti040** [comment]
1. The latest one, from Aminet.
2. No GuiGfx is **checked and disabled** in Settings.
3. Screenshots, in IFF format are correctly rendered by Multiview.
4. I got screenshots in each game, so that's not an issue.

**Comment by walkero-gr** [comment]
@arti040 Thank you for your message. 
If you have the gfx library usage disabled in iGame preferences, this means that it uses datatypes to display the images. Is there a possibility you run out of memory?

Can you please give me some info on your system, so that I can make an emulated version of it and try to reproduce it?
1. What hardware are you using? Accelerator? Ram?
2. Which version of the OS are you using?
3. Which version of MUI are you using?
4. Are you using some kind of OS distibution, like ClassicWB, BetterWB etc?
5. How many games does your list have?

**Comment by arti040** [comment]
1. A600 + 8MB FastRAM
2. 3.1
3. 3.8
4. [http://cleanwb.ppa.pl](http://cleanwb.ppa.pl)
5. 284 entries.

PS: https://winterland.no-ip.org/igame/files/guigfx_render_nofpu.lha seems to be corrupted. I'm not able to unpack it in any way...

**Comment by walkero-gr** [comment]
I am going to test it and let you know. Thank you for the info.

**Comment by walkero-gr** [comment]
@arti040 I think I found the issue. 

I guess you are using the 68000 version of CleanWB, right?
It is interesting how a package that targets 68000 has 020 versions for numerous libraries.
The problem with that is that the MCC_NList and the MCC_Texteditor are 020 versions, so, even they are installed they cannot open for your system.

You need to install in MUI:Libs/mui the following, and replace the old ones
http://aminet.net/package/dev/mui/MCC_TextEditor_68k
http://aminet.net/package/dev/mui/MCC_NList-0.102_68k

When I did the above iGame worked fine. Here is a screenshot of it, showing the intro image of iGame.

![Screenshot from 2023-11-14 22-25-06](https://github.com/MrZammler/iGame/assets/14014909/d202f9cd-a918-46a4-8dc4-6ed57e1d09da)

**Comment by walkero-gr** [comment]
@arti040  You can find the guigfx_render_nofpu.lha below

https://github.com/MrZammler/iGame/blob/master/guigfx_render_nofpu.lha

**Comment by arti040** [comment]
Oh! Nice catch. Thanks for bug-reporting my CleanWB then! :D I'll fix that. Sorry for inconvenience.

**Comment by walkero-gr** [comment]
Oh, I didn't know you were developing it. Thank you for including the iGame in your distribution.
When you select a game, this should start using WHDLoad. I guess the games work fine if started outside iGame, right?
Unfortunately, until the 27th I am away from my development system, and I can't test it further. If you can wait until, then I will have a look.

**Comment by arti040** [comment]
Actually, I'm trying to include it :) but experienced this problem in 68000. Games started directly work fine. Sure, can wait.

**Comment by walkero-gr** [comment]
Let's work on this together and see how to improve it for your users. I will contact you again as soon as I test it.

**Comment by arti040** [comment]
This is what I discovered today:
- replacing MUI classes and libraries with 68000 version makes screenshot loading to fail on every CPU...
- coping NList.mcc to MUI:Libs folder makes two things:
 * it is finally being properly loaded and SnoopDos does not indicates it as failed (however, location for this file is strange now, IMHO).
 * I'm able now to tick No GuiGfx off and screenshots start to being loaded on 020+ but still fail on 68000.

**Comment by walkero-gr** [comment]
The MUI MCC libs should be installed in MUI:Libs/mui. Maybe you keep same libraries in different version in two places and that makes applications crash.

**Comment by arti040** [comment]
Nope. All is right. Just program seems to not see NList.mcc in MUI:Libs/mui... 
Will back to it tomorrow.

**Comment by walkero-gr** [comment]
@arti040 Sorry I didn't get back to you earlier. If the program doesn't see the library, run the MUI window and check if MUI itself sees them. If not, then probably you have the wrong libraries for the CPU.

If you have a newer version of your package you would like me to test, please let me know where to get it.

**Comment by arti040** [comment]
OK! Will check it today and let you know.

**Comment by arti040** [comment]
MUI does not see NList.ncc class ;-)

**Comment by walkero-gr** [comment]
If I recall correct the one that I nstalled was http://aminet.net/package/dev/mui/MCC_NList-0.102_68k

**Comment by arti040** [comment]
There is kinda progress - now GUIGFX is still greyed out in Prefs but iGame no crashes at least anymore. However, instead of screenshot of the game I'm getting empty BOOPSI gadget.

**Comment by walkero-gr** [comment]
@arti040 GuiGFX is disabled because some library is missing. I  would suggest to not try to use guigfx because itself is problematic.
